dev-docs/source/Architecture.rst

```diff
@@ -0,0 +1,194 @@
+workspace objects, instrument objects, algorithm interfaces,
+along with a set of plugin libraries that are loaded at runtime to register
+new capabilities such as algorithms for data treatment.
+
+A Python package to the C++ libraries, called ``mantid``, is defined in
+``Framework/PythonInterface`` and
+gives access to the C++-framework code from Python.
+The Python package is the primary mode of access to the C++ libraries
+for external entities.
+Assuming Python paths have been setup correctly, the ``mantid`` package
+is to be treated like any other package and can be used in vanilla
+Python or any interactive Python environment,
+including `Jupyter notebooks <jupyter_>`_.
+
+
+While the C++ API could be used directly by external parties this is
+not a supported method of interaction and API stability of the C++
+layer is not guaranteed.
+
+mantidqt
+########
+
+.. figure:: /images/architecture-mantidqt.png
+   :scale: 85%
+   :align: center
+   :alt: Component diagram of mantidqt container
+
+   Container diagram for mantidqt Python package
+
+``mantidqt`` is a Python package defined to house a collection of Qt widgets
+that sit on top of ``mantid`` functionality.
+They provide a reusable set of widgets that can be used across multiple
+interfaces to give a consistent user experience when presenting given
+``mantid`` functionality to users. One example of such a widget is the
+``mantidqt.widgets.filefinder``, which sits on top of the framework
+``FileFinder``, and presents GUI elements to pass through and run the
+the framework logic.
+Providing a reusable widget gives a consistent user experience across
+multiple interfaces/applications when displaying ``mantid`` features to users.
+
+The package contains basic widgets, such as the aforementioned ``filefinder``,
+but also more advanced widgets such as
+:ref:`slice viewer <sliceviewer>` and
+the OpenGL-based :ref:`instrument viewer <InstrumentViewer>`,
+which combine many features of the framework together to provide advanced
+visualization to users.
+
+The package contains a mixture of C++ and Python code.
+It is defined in the ``qt/python/mantidqt`` directory of the source tree.
+Widgets written in C++ use the `sip <sip-ref_>`_ system to export the
+C++ classes to a Python module.
+The raw exported modules are all prefixed with an underscore,
+e.g. ``_common``,
+and classes are pulled into a pure-python module to decouple
+the C++ implementations from users.
+This allows for flexibility with the layout of the C++ classes/libraries
+without impacting users in the future.
+Clients of ``mantidqt`` should never import an underscored module directly.
+
+workbench
+#########
+
+.. figure:: /images/architecture-workbench.png
+   :scale: 85%
+   :align: center
+   :alt: Component diagram of workbench container
+
+   Container diagram for workbench application
+
+``workbench`` is an application written in pure Python in the ``qt/applications/workbench``
+directory of the source.
```

Comment:
Reading through we don't explicitly state that Workbench is our main "entrypoint" which allows users to do the majority of their work from a GUI perspective (that isn't an external GUI). I'm not sure if we want to state it here alongside the library section, or above where we introduce this?

```suggestion
directory of the source. It is intended to be the main interface provided by the project and allows access to the customized interfaces as described in `Container<container_>`.
```
Feel free to adapt this further as I'm not 100% how to phrase it

Reply:
Yeah this is a good point. Developers reading this should have an understanding of how the user community uses things.